packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart

```diff
@@ -223,7 +223,7 @@ class FirebaseOptions {
 
   @override
   int get hashCode {
-    return hashObjects(asMap.entries);
+    return asMap.toString().hashCode;
```

Comment:
This is a pretty expensive operation, first converting the object to a map and then running `toString()`. The `hashCode` is supposed to be computed significantly faster than true equality.

It would be much faster to compute the hash manually by looking at the properties, like many of the built-in types in the Flutter framework.

```suggestion
    return hashValues(
      apiKey ?? googleAppID,
      appId,
      messagingSenderId ?? gcmSenderID,
      projectId ?? projectID,
      authDomain,
      databaseURL,
      storageBucket,
      measurementId,
      trackingId ?? trackingID,
      deepLinkURLScheme,
      androidClientId ?? clientID,
      iosClientId ?? clientID,
      iosBundleId ?? bundleID,
      appGroupId,
    );
```

Reply:
Thanks, I agree 👍